src/App.tsx

```diff
@@ -1,61 +1,112 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function findUserById(id: number) {
+  return usersFromServer.find(user => user.id === id) || null;
+}
+
+const todoList = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: findUserById(todo.userId),
+  }
+));
 
 export const App = () => {
+  const [titleForNewTodos, setTitleForNewTodos] = useState('');
+  const [userIdForNewTodos, setUserIdForNewTodos] = useState(0);
+  const [newTodos, setNewTodos] = useState(todoList);
+  const [invalidTitle, setInvalidTitle] = useState(false);
+  const [invalidUser, setInvalidUser] = useState(false);
+
+  function getMaxId(todos: Todo[]): number {
```

Comment:
Consider moving `getMaxId` function to a separate file and exporting it as a utility function. It will make the code more modular and reusable. 🧩